### DISCUSSION TITLE
Content headers and/or raw formatting?

### DISCUSSION THREAD

**Discussion by kylekartan** [microfeed/microfeed#153] [Q&A] [answered]
Hopefully this repo is still being watched as it is a great project!
(let me preface that I am not a developer, just an IT guy who sucks at coding)

I'm having trouble with some clients/apps (looking at you Wix) being able to subscribe/parse the RSS URL properly. I was expecting the RSS feed to be in raw XML format but it renders nice and pretty as HTML. I've checked the headers and it's definitely coming in as content-type text/html.

I ran in to a similar issue when trying to parse the JSON (using another CF worker). It seems to work ok, but then throws errors because the JSON is coming in as text/html instead of application/json.

Is there a way to render a raw XML/JSON/RSS or am I misunderstanding the issue and how different content types work?

thanks

**Comment by wenbinf** [comment, marked as answer]
This is raw XML.

The reason it appears as a nicely styled webpage in the browser is due to the xml-stylesheet directive, which applies styling to the raw XML. You can find this on the second line of the RSS feed code:

https://www.microfeed.org/rss

<img width="728" alt="Screenshot 2024-11-06 at 10 56 44 AM" src="https://github.com/user-attachments/assets/e28b727b-3679-41c0-bd9f-89a5922dd1ed">

**Reply by kylekartan** [reply]
You're absolutely right, for the life of me I can't seem to get any 3rd party tools to integrate with the feed so I thought there was an error somewhere else. Spent like 4 hours trying to make a CF worker to convert the JSON feed to XML in JS. 

Thanks
